foundry.md

```diff
@@ -145,6 +157,73 @@ This rule then takes the account using `#asWord(#range(LM, ARGSTART +Int 4, 32)`
          </account>
 ```
 
+#### `expectRevert` - The next call is expected to fail
+
+In addition, cheatcodes should be ignored during the execution of the `expectRevert` cheatcode.
+
+```k
+    rule [call.expectRevert]:
+         <k> CALL _ CHEAT_ADDR 0 ARGSTART _ARGWIDTH 0 0 => 1 ~> #push ... </k>
+         <output> _ => .ByteArray </output>
+         <localMem> LM </localMem>
+         <expectRevert> false => true </expectRevert>
+         <expectedBytes> _ => .ByteArray </expectedBytes>
+      requires CHEAT_ADDR ==Int #address(FoundryCheat)
+       andBool #asWord(#range(LM, ARGSTART, 4)) ==Int 4102309908 // selector ( "expectRevert()" )
+      [priority(40)]
+
+    rule [call.expectRevertBytes]:
+         <k> CALL _ CHEAT_ADDR 0 ARGSTART _ARGWIDTH 0 0 => 1 ~> #push ... </k>
+         <output> _ => .ByteArray </output>
+         <localMem> LM </localMem>
+         <expectRevert> false => true </expectRevert>
+         <expectedBytes> _ => #range(LM, ARGSTART +Int 4, 32) </expectedBytes>
+      requires CHEAT_ADDR ==Int #address(FoundryCheat)
+       andBool #asWord(#range(LM, ARGSTART, 4)) ==Int 4069379763 // selector ( "expectRevert(bytes)" )
+      [priority(40)]
+
+    rule [call.expectRevertBytes4]:
+         <k> CALL _ CHEAT_ADDR 0 ARGSTART _ARGWIDTH 0 0 => 1 ~> #push ... </k>
+         <output> _ => .ByteArray </output>
+         <localMem> LM </localMem>
+         <expectRevert> false => true </expectRevert>
+         <expectedBytes> _ => #range(LM, ARGSTART +Int 4, 4) </expectedBytes>
+      requires CHEAT_ADDR ==Int #address(FoundryCheat)
+       andBool #asWord(#range(LM, ARGSTART, 4)) ==Int 3273568480 // selector ( "expectRevert(bytes4)" )
+      [priority(40)]
+
+    rule [call.setEndExpectRevert]:
+         <k> CALL _ ADDR _ _ _ _ _ ~> (. => #endExpectRevert) ~> KITEM ... </k>
+         <output> _ => .ByteArray </output>
+         <expectRevert> true </expectRevert>
+      requires ADDR =/=Int #address(FoundryCheat)
+       andBool KITEM =/=K #endExpectRevert
+      [priority(40)]
```

Comment:
I am not understanding this rule very well. This is not a rule for a cheatcode right? 
From my understanding this rule will be applied after the first revert call happens? But how? I am not getting the flow. Can you please help me to understand?